server/src/main/java/org/opensearch/common/cache/tier/TieredCacheSpilloverStrategyService.java

```diff
@@ -23,8 +28,9 @@
  * cache items to disk tier cache.
  * @param <K> Type of key
  * @param <V> Type of value
+ * @param <W> Type that V can be unpacked into for inspection by policies. Can be the same as V.
  */
-public class TieredCacheSpilloverStrategyService<K, V> implements TieredCacheService<K, V>, RemovalListener<K, V> {
+public class TieredCacheSpilloverStrategyService<K, V, W> implements TieredCacheService<K, V, W>, RemovalListener<K, V> {
```

Comment:
Let's remove this W. Make's it messy. 
By changes suggested in IRC, u can have CacheTierPolicy<V>
Pass the function directly into DiskTookTimePolicy so that it can extract took time from ByteReference as suggested.